docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,240 @@
+# - 3 bypass diodes half-cut cells:
+#   - 2 blocks in portrait
+#   - 3 blocks in landscape
+#
+# .. figure:: ../../_images/PV_module_layout_cesardd.jpg
+#    :align: center
+#    :width: 75%
+#    :alt: Normal and half-cut cells module layouts
+#
+#    Left: common module layout. Right: half-cut cells module layout.
+#    Each module has three bypass diodes. On the left, they connect cell
+#    columns 1-2, 2-3 & 3-4. On the right, they connect cell rows 1-2, 3-4 &
+#    5-6.
+#    *Source: César Domínguez. CC BY-SA 4.0, Wikimedia Commons*
+#
+# In the upper image, each orange U-shaped section of the string is a block.
+# By symmetry, the yellow inverted-U's of the subcircuit are also blocks.
+# For this reason, the half-cut cell modules have 6 blocks in total: two along
+# the longest side and three along the shortest side.
+
+blocks_per_module = {
+    "1 bypass diode": 1,
+    "3 bypass diodes": 3,
+    "3 bypass diodes half-cut, portrait": 2,
+    "3 bypass diodes half-cut, landscape": 3,
+}
+
+# Calculate the number of shaded blocks during the day
+shaded_blocks_per_module = {
+    k: np.ceil(blocks_N * shaded_fraction)
+    for k, blocks_N in blocks_per_module.items()
+}
+
+# %%
+# Results
+# -------
+# Now that we have the number of shaded blocks for each module configuration,
+# we can apply the model and estimate the power loss due to shading.
+#
+# Note this model is not linear with the shaded blocks ratio, so there is a
+# difference between applying it to just a module or a whole row.
+
+shade_factor_per_module = {
+    k: pvlib.shading.direct_martinez(
+        shaded_fraction, module_shaded_blocks, blocks_per_module[k]
+    )
+    for k, module_shaded_blocks in shaded_blocks_per_module.items()
+}
+
+shade_factor_per_row = {
+    k: pvlib.shading.direct_martinez(
+        shaded_fraction,
+        module_shaded_blocks * N_modules_per_row,
+        blocks_per_module[k] * N_modules_per_row,
+    )
+    for k, module_shaded_blocks in shaded_blocks_per_module.items()
+}
+
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
+fig.suptitle("Martinez power correction factor due to shading")
+for k, shade_factor in shade_factor_per_module.items():
+    linestyle = "--" if k == "3 bypass diodes half-cut, landscape" else "-"
+    ax1.plot(times, shade_factor, label=k, linestyle=linestyle)
+ax1.legend()
+ax1.grid()
+ax1.set_xlabel("Time")
+ax1.xaxis.set_major_formatter(
+    ConciseDateFormatter("%H:%M", tz="Europe/Madrid")
+)
+ax1.set_ylabel("Power correction factor")
+ax1.set_title("Per module")
+
+for k, shade_factor in shade_factor_per_row.items():
+    linestyle = "--" if k == "3 bypass diodes half-cut, landscape" else "-"
+    ax2.plot(times, shade_factor, label=k, linestyle=linestyle)
+ax2.legend()
+ax2.grid()
+ax2.set_xlabel("Time")
+ax2.xaxis.set_major_formatter(
+    ConciseDateFormatter("%H:%M", tz="Europe/Madrid")
+)
+ax2.set_ylabel("Power correction factor")
+ax2.set_title("Per row")
+fig.tight_layout()
+fig.show()
```

Comment:
I guess the block plotting the figure could be hidden (or collapsed) for making the example more compact

Reply:
I prefer leaving it as is, so in case of doubt it can be consulted without digging into the repo files. If I knew how to collapse it I would do it, but I think that's impossible. In any case, I see both pros and cons regarding that suggestion, so feedback is more than welcome @AdamRJensen @kandersolar.